### PR TITLE
Update to latest geth and parity

### PR DESCRIPTION
**Pull request by palango**
## Description

Fixes: #5386 

Skipped test tracked in https://github.com/raiden-network/raiden/issues/5606
## PR review check list

Quality check list that cannot be automatically verified.

- Safety
    - [ ] The changes respect the necessary conditions for safety (https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-constraints)
-  Code quality
    - [ ] Error conditions are handled
    - [ ] Exceptions are propagated to the correct parent greenlet
    - [ ] Exceptions are correctly classified as recoverable or unrecoverable
- Compatibility
    - [ ] State changes are forward compatible
    - [ ] Transport messages are backwards and forward compatible
- Commits
    - [ ] Have good messages
    - [ ] Squashed unecessary commits
- If it's a bug fix:
    - Regression test for the bug
        - [ ] Properly covers the bug
        - [ ] If an integration test is used, it could not be written as a unit test
- Documentation
    - [ ] A new CLI flag was introduced, is there documentation that explains usage?
- Specs
    - [ ] If this is a protocol change, are the specs updated accordingly? If so, please link PR from the specs repo.
- Is it a user facing feature/bug fix?
    - [ ] Changelog entry has been added

### CODE REVIEW

**Comment by Dominik1999** [comment]
rebase PR, figure out why it is flaky, fix flakiness and try it again

File: raiden/constants.py [hunk 1]
```diff
@@ -165,10 +165,10 @@ class RoutingMode(Enum):
 
 
 # Keep in sync with .circleci/config.yaml
-HIGHEST_SUPPORTED_GETH_VERSION = "1.9.2"
+HIGHEST_SUPPORTED_GETH_VERSION = "1.9.8"
```
Comment (palango) [inline raiden/constants.py]: This needs to be fixed

**Comment by palango** [comment]
I dig a bit more. Seems like `test_remote_transaction_with_zero_gasprice_is_not_mined` is the only test that uses two blockchain nodes
```py
@pytest.mark.parametrize("blockchain_number_of_nodes", [2])
```

And for some reason the second, no-miner client isn't connecting properly to the first.

**Comment by hackaugusto** [comment]
I managed to reproduce the error locally, it is flaky.

For some reason the failing run didn't start to sync, i.e. it is missing this log line:

> INFO [12-13|12:14:26.000] Block synchronisation started 

Logs for the non-miners: 
https://gist.github.com/hackaugusto/dc77ae92e7a1332f1f30139c52a591cd

Logs for the miners:
https://gist.github.com/hackaugusto/17c1ec4d522aff0ba87c192a711d8bea

My first thought is a race condition were the non-miner starts the p2p layer before the miner

File: raiden/constants.py (hunk 1 above)
Comment (kilrau) [reply raiden/constants.py]: To 1.9.9, yes. But didn't you merge a fix that ignores minor versions?

Comment (palango) [reply raiden/constants.py]: Yes, it should run with higher versions anyways.

File: .circleci/config.yml [hunk 2]
```diff
@@ -39,16 +39,16 @@ templates:
...
+    GETH_URL_LINUX: 'https://gethstore.blob.core.windows.net/builds/geth-linux-amd64-1.9.9-01744997.tar.gz'
+    GETH_MD5_LINUX: 'de1b31439dc6697fcbe2d729d2080d22'
+    GETH_URL_MACOS: 'https://gethstore.blob.core.windows.net/builds/geth-darwin-amd64-1.9.9-01744997.tar.gz'
+    GETH_MD5_MACOS: '9ebd527e9d0ac97435f52fa6dc7d7bdb'
+    GETH_VERSION: '1.9.9'
+    PARITY_URL_LINUX: 'https://releases.parity.io/ethereum/v2.5.11/x86_64-unknown-linux-gnu/parity'
+    PARITY_SHA256_LINUX: '20fba204357758008a4f2fac29da552e79863980424a98d1c20fc228690e33a4'
+    PARITY_URL_MACOS: 'https://releases.parity.io/ethereum/v2.5.11/x86_64-apple-darwin/parity'
+    PARITY_SHA256_MACOS: '829de731e081fe4eccd5b1ae70297b2ded7f9e712d36ed63aa6e8b1066f35073'
+    PARITY_VERSION: 'v2.5.11'
```
Comment (LefterisJP) [inline .circleci/config.yml]: The latest parity ethereum is 2.5.13 ->  https://github.com/paritytech/parity-ethereum/releases/tag/v2.5.13

File: raiden/tests/integration/rpc/assumptions/test_rpc_gas_price_assumptions.py [hunk 3]
```diff
@@ -155,7 +155,11 @@ def test_local_transaction_with_zero_gasprice_is_mined(deploy_client: JSONRPCCli
 
 @pytest.mark.parametrize("blockchain_number_of_nodes", [2])
 def test_remote_transaction_with_zero_gasprice_is_not_mined(
-    web3: Web3, deploy_key: PrivateKey, blockchain_rpc_ports: List[Port], blockchain_type: str
+    web3: Web3,
+    deploy_key: PrivateKey,
+    blockchain_rpc_ports: List[Port],
+    blockchain_type: str,
+    skip_if_not_parity,
```
Comment (LefterisJP) [inline raiden/tests/integration/rpc/assumptions/test_rpc_gas_price_assumptions.py]: Can you also make an issue and link it here with a comment?

Comment (palango) [reply raiden/tests/integration/rpc/assumptions/test_rpc_gas_price_assumptions.py]: Done

Comment (LefterisJP) [reply raiden/tests/integration/rpc/assumptions/test_rpc_gas_price_assumptions.py]: I meant a comment in the code. Generally if I see a skip in the code I want to see an issue next to it.

**Pull request merged**